.github/PULL_REQUEST_TEMPLATE/new_resource.md

```diff
@@ -14,6 +14,7 @@ about: For when you're adding a new resource or datasource
 - [ ] Summary includes `Closes <issue>`, if available
 - [ ] The [contributing guide](https://github.com/PrefectHQ/terraform-provider-prefect/blob/main/_about/CONTRIBUTING.md) has been read
 - [ ] Documentation is added (generated by `make docs` from source code)
+      - When applicable, provide a link back to the relevant page in the [Prefect documentation site](https://docs.prefect.io).
```

Comment:
Added a reminder here to hopefully include this consistently in future additions.